external/fv3fit/fv3fit/_shared/taper_function.py

```diff
@@ -0,0 +1,106 @@
+import abc
+import numpy as np
+import xarray as xr
+
+from fv3fit._shared.novelty_detector import NoveltyDetector
+
+
+class TaperFunction(abc.ABC):
+    @abc.abstractmethod
+    def get_taper_value(self, novelty_score: xr.DataArray) -> xr.DataArray:
```

Comment:
This is an interface with a single method. Often this implies that it should just be a function instead. Would it be possible to implement all the TaperFunctions as actual functions? They look nice and tidy that way. For example:

```
def taper_mask(novelty_score, cutoff):
    return xr.where(novelty_score > cutoff, 0, 1)

def taper_ramped(novelty_score, max, min):
    unclipped = (max - novelty_score) / (max - min)
    return np.clip(unclipped, 0, 1)
```


Then you can put all the instantiation logic into a single factory function:

```
def get_taper_function(config: dict) -> Callable[[xr.DataArray], xr.DataArray]:
    if "cutoff" in config:
        return lambda x: taper_mask(x, **config)
    elif "ramp_min" in config:
        return lambda x: taper_ramped(x, **config)
    ...
```

The advantage of this is that there is less coupling between the argument binding/config and the implementations themselves. Also, you don't need to worry about the name `taper_func.get_taper_value(x)` wherever you use it...which is just more typing and reasons to change compared to `taper_func(x)` IMO.

Reply:
Thank you, this is a great suggestion! Working on it now.